packages/block-editor/src/components/grid-interactivity/grid-visualizer.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __experimentalUseDropZone as useDropZone } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { __unstableUseBlockElement as useBlockElement } from '../block-list/use-block-props/use-block-refs';
+import BlockPopoverCover from '../block-popover/cover';
+import { getComputedCSS } from './utils';
+import { parseDropEvent } from '../use-on-block-drop';
+import { store as blockEditorStore } from '../../store';
+
+export function GridVisualizer( { clientId } ) {
+	const blockElement = useBlockElement( clientId );
+	if ( ! blockElement ) {
+		return null;
+	}
+	return (
+		<GridVisualizerGrid
+			clientId={ clientId }
+			blockElement={ blockElement }
+		/>
+	);
+}
+
+function getGridInfo( blockElement ) {
+	const gridTemplateColumns = getComputedCSS(
+		blockElement,
+		'grid-template-columns'
+	);
+	const gridTemplateRows = getComputedCSS(
+		blockElement,
+		'grid-template-rows'
+	);
+	const numColumns = gridTemplateColumns.split( ' ' ).length;
+	const numRows = gridTemplateRows.split( ' ' ).length;
+	const numItems = numColumns * numRows;
+	return {
+		numColumns,
+		numRows,
+		numItems,
+		style: {
+			gridTemplateColumns,
+			gridTemplateRows,
+			gap: getComputedCSS( blockElement, 'gap' ),
+			padding: getComputedCSS( blockElement, 'padding' ),
+		},
+	};
+}
+
+function range( start, length ) {
+	return Array.from( { length }, ( _, i ) => start + i );
+}
+
+function GridVisualizerGrid( { clientId, blockElement } ) {
+	const [ gridInfo, setGridInfo ] = useState( () =>
+		getGridInfo( blockElement )
```

Comment:
For the new kid: this is so the visualizer matches the underlying grid when resizing the window? Works really well. Nice 🤟🏻 

Feel free to ignore this opinion: "Info" sounds to me like something I'd ask for at a truck stop on my way to an obscure holiday township with no mobile reception.

Just jokes. What about 'properties'? E.g.,  `getBlockElementGridProperties`

Reply:
It _does_ handle window resizing but that's secondary. The primary purpose is so that we recalculate the grid lines when the grid or its children change size e.g. if the user changes the number of columns in the inspector. 

I didn't want to use the words _attributes_ or _properties_ because I didn't want to imply that we're returning HTML attributes, React props, or CSS properties. I think a generic word like _data_ or _info_ is right. But idk happy to do what you say here 🤷‍♂️

Reply:
All good! Just a passing comment with as much value as it cost in the first place = `$0` 😄

Reply:
Good to ask clarifying questions, it means they're answered for future readers too 🙂